File: LeetCode/28/solution.ts
function strStr(haystack: string, needle: string): number {
    const needle_start: string = needle[0];

    for (let i = 0; i < haystack.length; i++) {
        if (needle_start === haystack[i]) {
            if (needle === haystack.slice(i, (i + needle.length))) {
                return i;
            }
        }
    };

    return -1;
};

const ICE = strStr("sadbutsad", "sad");
console.log(ICE)

File: LeetCode/28/solution.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        needle_start = needle[0]

        for i in range(0, len(haystack)):
            if needle_start == haystack[i]:
                if needle == haystack[i : (i + len(needle))]:
                    return i
        
        return -1
    
ice = Solution()
result = ice.strStr("sadbutsad", "sad")
print(f"\033[92m")
print(result)